src/tsickle_compiler_host.ts

```diff
@@ -0,0 +1,259 @@
+import * as path from 'path';
+import {SourceMapConsumer, SourceMapGenerator} from 'source-map';
+import * as ts from 'typescript';
+
+import {convertDecorators} from './decorator-annotator';
+import {processES5} from './es5processor';
+import {ModulesManifest} from './modules_manifest';
+import {annotate} from './tsickle';
+
+export enum Pass {
+  None,
+  DecoratorDownlevel,
+  Tsickle
+}
+
+export interface TsickleCompilerHostOptions {
+  googmodule: boolean;
+  es5Mode: boolean;
+  tsickleTyped: boolean;
+  prelude: string;
+}
+
+export interface TsickleEnvironment {
+  /**
+   * If true, tsickle and decorator downlevel processing will be skipped for
+   * that file.
+   */
+  shouldSkipTsickleProcessing: (fileName: string) => boolean;
```

Comment:
Are you intentionally using properties that are function valued instead of methods? I.e. why not `interface X { foo(bar: baz): bam; }`?